java-client/src/main/java/co/elastic/clients/transport/rest5_client/low_level/Rest5ClientBuilder.java

```diff
@@ -320,6 +321,36 @@ public Rest5ClientBuilder setMetaHeaderEnabled(boolean metadataEnabled) {
         return this;
     }
 
+    /**
+     * Sets the {@link HttpClientConfigCallback} to be used to customize http client configuration
+     *
+     * @throws NullPointerException if {@code httpClientConfigCallback} is {@code null}.
+     */
+    public Rest5ClientBuilder setHttpClientConfigCallback(HttpClientConfigCallback httpClientConfigCallback) {
+        Objects.requireNonNull(httpClientConfigCallback, "httpClientConfigCallback must not be null");
+        this.httpClientConfigCallback = httpClientConfigCallback;
+        return this;
+    }
+
+    /**
+     * Callback used to customize the {@link CloseableHttpAsyncClient} instance used by a
+     * {@link Rest5Client} instance.
+     * Allows to customize default {@link RequestConfig} being set to the client and any parameter that
+     * can be set through {@link HttpAsyncClientBuilder}
+     */
+    public interface HttpClientConfigCallback {
+        /**
+         * Allows to customize the {@link CloseableHttpAsyncClient} being created and used by the
+         * {@link Rest5Client}.
+         * Commonly used to customize {@link HttpAsyncClientBuilder} without losing any other useful default
+         * value that the {@link Rest5ClientBuilder} internally sets, except if RequestConfig,
+         * ConnectionConfig
+         * and ConnectionManager are set through this callback.
+         * In those cases, all default values set by the {@link Rest5ClientBuilder} are lost.
+         */
+        HttpAsyncClientBuilder customizeHttpClient(HttpAsyncClientBuilder httpClientBuilder);
+    }
```

Comment:
This construct in the legacy RestClient is inherited from pre-Java 8 times (yeah, Elasticsearch is that old) and IMHO we can just replace that with a `Consumer<HttpAsyncClientBuilder>`.